docs/content/en/docs/clustermgmt/cluster-upgrades/vsphere-and-cloudstack-upgrades.md

```diff
@@ -15,6 +15,11 @@ The upgrade command also upgrades core components of EKS Anywhere and lets the u
 
 **Upgrades should never be run from ephemeral nodes (short-lived systems that spin up and down on a regular basis). If an upgrade fails, it is very important not to delete the Docker containers running the KinD bootstrap cluster. During an upgrade, the bootstrap cluster contains critical EKS Anywhere components. If it is deleted after a failed upgrade, they cannot be recovered.**
 
+*Please make sure to have sufficient available IP addresses in your DHCP pool from VMs counts and maxSurge config. For create operation, each VM needs 1 IP. For upgrade and scale operation, control plane and works need just 1 extra IP (total not per node) due to rolling upgrade strategy. Each external ETCD needs 1 extra IP address (ex: 3 ETCD nodes would require 3 more IP addresses) because etcdadm-controller needs to create all new ETCD machines before removing the old machines. You would also need additional IPs equal to the number used for maxSurge. After calculating the required IPs, please make sure your environment has enough available IPs before upgrade operation.
+  Example 1, to create a cluster with 1 control plane node, 2 worker node and 3 unstacked ETCD nodes, you'll need at least 6 (1+2+3) available IPs. To upgrade the same cluster with default maxSurge (0), you'll need 1 (1+0+0) additional available IPs.
+  Example 2, to upgrade a cluster with 1 control plane node, 2 worker node and 3 stacked (external) ETCD nodes, you'll need at least 4 (1+3+0) additional available IPs.
+  Example 3, to upgrade a cluster with 1 control plane node, 2 worker node and 3 stacked (external) ETCD nodes, with maxSurge set to 2, you'll need at least 6 (1+3+2) additional available IPs.
+
```

Comment:
We might want examples for CP nodes > 1. So users see how to handle that calculation.